src/fsharp/CheckComputationExpressions.fs

```diff
@@ -1465,7 +1466,9 @@ let TcComputationExpression cenv env overallTy tpenv (mWhole, interpExpr: Expr,
                     if isNil (TryFindIntrinsicOrExtensionMethInfo ResultCollectionSettings.AtMostOneResult cenv env m ad "Return" builderTy) then
                         SynExpr.ImplicitZero m
                     else
-                        SynExpr.YieldOrReturn((false, true), SynExpr.Const(SynConst.Unit, m), m)
+                        match TryFindIntrinsicOrExtensionMethInfo ResultCollectionSettings.AtMostOneResult cenv env m ad "Zero" builderTy with
```

Comment:
This change is specified in [the TaskBuilder RFC](https://github.com/fsharp/fslang-design/blob/main/RFCs/FS-1097-task-builder.md#feature-respecting-zero-methods-in-computation-expressions), but in that section I'm having a hard time visualizing a situation that currently fails. Could you give an example there to help clarify?

Reply:
The difference is that a `Zero()` is produced instead of a `Return(())` on the empty branch

The `Return` stores the result in the state machine and pins down the overall type produced by the state machine.  Zero just continues successfully.

```
        /// Used to represent no-ops like the implicit empty "else" branch of an "if" expression.
        [<DefaultValue>]
        member inline _.Zero() : TaskCode<'TOverall, unit> = ResumableCode.Zero()

        member inline _.Return (value: 'T) : TaskCode<'T, 'T> = 
            TaskCode<'T, _>(fun sm -> 
                sm.Data.Result <- value
                true)
```

Reply:
I'll add this to the RFC

Reply:
I think we should put a comment here describing why we are doing this. It's subtle but important.

Reply:
Nevermind, I see the comment, it's just above it at the top of the method. :)